modules/fedimint-ln-client/src/lib.rs

```diff
@@ -838,26 +838,29 @@ impl LightningClientModule {
             Some(active_gateway) => Ok(active_gateway.info),
             None => {
                 let gateways = self.fetch_registered_gateways().await?;
+                let num_gateways = gateways.len();
 
                 let vetted = gateways
                     .clone()
                     .into_iter()
                     .filter(|g| g.vetted)
                     .collect::<Vec<_>>();
                 if !vetted.is_empty() {
-                    debug!("Choosing a vetted gateway");
-                    vetted
+                    let chosen_gateway = vetted
                         .into_iter()
                         .map(|gw| gw.info)
                         .choose(&mut rand::thread_rng())
-                        .ok_or(anyhow::anyhow!("Could not choose a vetted gateway"))
+                        .expect("vetted is non-empty");
```

Comment:
But this error message  will  log  when there is panic due to  some bug or unexpected condition -> so IMO , this should not be the message.

Reply:
feels correct to me, given we just check vetted.is_empty just above